packages/flutter/lib/src/material/text_selection.dart

```diff
@@ -119,8 +119,10 @@ class _TextSelectionHandlePainter extends CustomPainter {
   @override
   void paint(Canvas canvas, Size size) {
     final Paint paint = Paint()..color = color;
+    final Paint filterPaint = Paint()..blendMode = BlendMode.dstOut;
     final double radius = size.width/2.0;
     canvas.drawCircle(Offset(radius, radius), radius, paint);
+    canvas.drawRect(Rect.fromLTWH(0.0, 0.0, radius, radius), filterPaint);
     canvas.drawRect(Rect.fromLTWH(0.0, 0.0, radius, radius), paint);
```

Comment:
Do you really need two separate paints here?  Can you just add the blend mode to the existing paint and paint with that?

Reply:
Ah, I see.

I think it would be more efficient to just do this:

```dart
    final Paint paint = Paint()..color = color;
    final double radius = size.width/2.0;
    final Rect circle = Rect.fromCircle(center: Offset(radius, radius), radius: radius);
    final Rect point = Rect.fromLTWH(0.0, 0.0, radius, radius);
    final Path path = Path()..addOval(circle)..addRect(rect);
    canvas.drawPath(path, paint);
```

Then no blending or multiple draw calls.

Reply:
(for example, I think your dstOut blend mode could cause problems if someone removed a saveLayer somewhere)

Reply:
Thanks, I didn't know that I can acvieve this with ui.Path 🙂 
Yeah, `saveLayer` manipulations could crash my initial solution.